node/src/stream_handler_pool.rs

```diff
@@ -367,218 +370,352 @@ impl StreamHandlerPool {
-                    "connection for {} in progress, resubmitting {} bytes",
-                    peer_addr,
-                    msg.context.data.len()
-                );
-                let recipient = self
-                    .self_subs_opt
-                    .as_ref()
-                    .expect("StreamHandlerPool is unbound.")
-                    .node_query_response
-                    .clone();
-                // TODO FIXME revisit once SC-358/GH-96 is done (idea: create an actor for delaying messages?)
-                thread::spawn(move || {
-                    // to avoid getting into too-tight a resubmit loop, add a delay; in a separate thread, to avoid delaying other traffic
-                    thread::sleep(Duration::from_millis(100));
-                    recipient.try_send(msg).expect("StreamHandlerPool is dead");
-                });
-            }
+            Some(None) => self.delay_packet_for_opening_stream(msg, peer_addr, sw_key),
             None => {
                 if peer_addr.ip() == localhost() {
-                    error!(
-                        self.logger,
+                    return Err(format!(
                         "Local connection {:?} not found. Discarding {} bytes.",
                         peer_addr,
                         msg.context.data.len()
-                    );
-                    return;
+                    ));
+                };
+
+                self.stream_writers
+                    .insert(StreamWriterKey::from(peer_addr), None);
+
+                self.open_new_stream_and_recycle_message(msg, peer_addr, sw_key);
+            }
+        }
+
+        Ok(())
+    }
+
+    fn send_packet_on_open_stream(
+        &self,
+        msg: DispatcherNodeQueryResponse,
+        peer_addr: SocketAddr,
+        sw_key: StreamWriterKey,
+        tx_box: &dyn SenderWrapper<SequencedPacket>,
+    ) -> Result<bool, String> {
+        debug!(
+            self.logger,
+            "Found already-open stream to {} keyed by {}: using",
+            tx_box.peer_addr(),
+            sw_key
+        );
+        debug!(self.logger, "Masking {} bytes", msg.context.data.len());
+        let packet = if msg.context.sequence_number.is_none() {
+            let masquerader = self.traffic_analyzer.get_masquerader();
+            match masquerader.mask(msg.context.data.as_slice()) {
+                Ok(masked_data) => SequencedPacket::new(masked_data, 0, false),
+                Err(e) => {
+                    return Err(format!(
+                        "Masking failed for {}: {}. Discarding {} bytes.",
+                        peer_addr,
+                        e,
+                        msg.context.data.len()
+                    ));
                 }
+            }
+        } else {
+            SequencedPacket::from(&msg.context)
+        };
 
+        let packet_len = packet.data.len();
+        match tx_box.unbounded_send(packet) {
+            Err(e) => {
                 debug!(
```

Comment:
This should probably be an `error!` log, unless it turns out to happen very often, in which case it should be a `warning!` log.